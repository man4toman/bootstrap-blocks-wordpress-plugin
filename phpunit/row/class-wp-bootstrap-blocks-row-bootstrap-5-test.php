<?php
/**
 * Row tests.
 *
 * @package wp-bootstrap-blocks
 */

/**
 * Class WP_Bootstrap_Blocks_Row_Test
 */
class WP_Bootstrap_Blocks_Row_Bootstrap_5_Test extends WP_Bootstrap_Blocks_UnitTestCase {
	/**
	 * Name of current block.
	 *
	 * @var string $blockname
	 */
	protected $blockname = 'row';

	/**
	 * Runs before each test
	 */
	public function setUp() {
		parent::setUp();
		// Enable Bootstrap 5
		update_option( \WP_Bootstrap_Blocks\Settings::OPTION_PREFIX . 'bootstrap_version', 5 );
	}

	/**
	 * Runs after each test
	 */
	public function tearDown() {
		parent::tearDown();
		delete_option( \WP_Bootstrap_Blocks\Settings::OPTION_PREFIX . 'bootstrap_version' );
	}

	/**
	 * Tests row no gutters.
	 */
	public function test_row_no_gutters() {
		$variant = 'no-gutters-bootstrap-5';
		list( $expected, $actual ) = $this->get_block_output( $variant );
		$this->assertEquals( $expected, $actual );
	}

	/**
	 * Tests row horizontal gutters.
	 */
	public function test_row_horizontal_gutters() {
		$variant = 'horizontal-gutters';
		list( $expected, $actual ) = $this->get_block_output( $variant );
		$this->assertEquals( $expected, $actual );
	}

	/**
	 * Tests row vertical gutters.
	 */
	public function test_row_vertical_gutters() {
		$variant = 'vertical-gutters';
		list( $expected, $actual ) = $this->get_block_output( $variant );
		$this->assertEquals( $expected, $actual );
	}
}
