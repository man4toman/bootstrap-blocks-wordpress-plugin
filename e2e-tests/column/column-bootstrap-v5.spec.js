/**
 * WordPress dependencies
 */
import {
	activatePlugin,
	deactivatePlugin,
	createNewPost,
	selectBlockByClientId,
	getEditedPostContent,
} from '@wordpress/e2e-test-utils';
import { insertRowBlock } from '../row/row-helper';
import { getColumnBlocks } from './column-helper';
import {
	clickElementByText,
	getDataValuesOfElement,
	openSidebarPanelWithTitle,
} from '../helper';

describe( 'column block Bootstrap 5', () => {
	beforeAll( async () => {
		await activatePlugin( 'wp-bootstrap-blocks-test-bootstrap-v5' );
	} );

	afterAll( async () => {
		await deactivatePlugin( 'wp-bootstrap-blocks-test-bootstrap-v5' );
	} );

	beforeEach( async () => {
		await createNewPost();
	} );

	it( 'Should display xxl breakpoint options', async () => {
		await insertRowBlock();

		// Select first column block
		const columnBlocks = await getColumnBlocks();
		const firstColumnBlockClientId = columnBlocks[ 0 ].clientId;
		await selectBlockByClientId( firstColumnBlockClientId );
		await openSidebarPanelWithTitle( 'Column size' );

		// Xxl column count option should exist
		expect(
			await page.$x(
				'//label[@class="components-base-control__label"][contains(text(),"Xxl Column count")]'
			)
		).toHaveLength( 1 );

		// Xxl equal-width option should exist
		expect(
			await page.$x(
				'//label[@class="components-checkbox-control__label"][contains(text(),"Xxl equal-width")]'
			)
		).toHaveLength( 1 );
	} );

	it( 'Should be possible to select column size for xxl breakpoint', async () => {
		await insertRowBlock();

		// Select first column block
		const columnBlocks = await getColumnBlocks();
		const firstColumnBlockClientId = columnBlocks[ 0 ].clientId;
		await selectBlockByClientId( firstColumnBlockClientId );
		await openSidebarPanelWithTitle( 'Column size' );

		// Change column count
		await page.type( '[aria-label="Xxl Column count"]', '2' );
		const columnData = await getDataValuesOfElement(
			`#block-${ firstColumnBlockClientId }`
		);
		expect( columnData.sizeXxl ).toMatch( '2' );

		// Enable column equal-width
		await clickElementByText( 'label', 'Xxl equal-width' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );
} );
