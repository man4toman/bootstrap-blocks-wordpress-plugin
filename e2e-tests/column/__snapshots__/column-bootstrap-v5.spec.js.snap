// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`column block Bootstrap 5 Should be possible to select column size for xxl breakpoint 1`] = `
"<!-- wp:wp-bootstrap-blocks/row -->
<!-- wp:wp-bootstrap-blocks/column {\\"sizeXxl\\":2,\\"sizeMd\\":6,\\"equalWidthXxl\\":true} /-->

<!-- wp:wp-bootstrap-blocks/column {\\"sizeMd\\":6} /-->
<!-- /wp:wp-bootstrap-blocks/row -->"
`;
